conformance/utils/suite/suite.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"testing"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+)
+
+// ConformanceTestSuite defines the test suite used to run Gateway API
+// conformance tests.
+type ConformanceTestSuite struct {
+	Client           client.Client
+	RoundTripper     roundtripper.RoundTripper
+	GatewayClassName string
+	ControllerName   string
+	Debug            bool
+	Cleanup          bool
+}
+
+// New returns a new ConformanceTestSuite.
+func New(c client.Client, gcName string, debug, cleanup bool, roundTripper roundtripper.RoundTripper) *ConformanceTestSuite {
```

Comment:
Since new arguments may be added to this function, replacing the arguments of this function with a "SuiteOpts" struct could be better.

Reply:
Updated, thanks!